DETAILS.md

```diff
@@ -0,0 +1,115 @@
+# Details on Athens functionality.
+
+This document describes the various use cases that Athens supports or will
+support. Not everything here is implemented.
+
+[@bketelsen](https://github.com/bketelsen) and I spoke at length on Slack about
+how the Athens registry should work. I'm putting our discussion down on "paper"
+and adding some of my own detail (and some opinions) so we can grow this into
+a definitive document.
+
+# Where Will the Code Come From?
+
+Here's what was discussed on slack + various other forums:
+
+## Travis or Other CI systems (Webhook)
+
+After code passes tests, the publisher can hit a webhook to tell the registry to fetch
+code for the given tag from the given repository. The webhook will then do the following:
+
+* Check if the repository is already "known" (the owner of the repository will need to register it prior) - Fail if it isn't
+* Check if the given tag already exists (we want versions to be immutable) - Fail if it does
+* Download the code from the given tag and store it in a CDN
+
+We may need to require users to be authorized to hit the webhooks.
+
+## Manual Uploads
+
+This works similarly to the webhook flow, except a user would upload their module --
+including code -- manually to the registry. When they upload, they'll need to
+specify the module name and version, and obviously include the zip file with their
+source code.
+
+The zip file will need to have the `go.mod` file in it, and the server will
+parse it out.
```

Comment:
this may need to be two separate files uploaded - need to verify spec.

Reply:
so this would only be for an upload protocol. for the download protocol, the `go.mod` definitely needs to be available outside of the zip